packages/addon-dev/src/rollup-gjs-plugin.ts

```diff
@@ -0,0 +1,71 @@
+import { createFilter } from '@rollup/pluginutils';
+import type { Plugin, PluginContext, ResolvedId } from 'rollup';
+import { readFileSync } from 'fs';
+import { Preprocessor } from 'content-tag';
+
+const PLUGIN_NAME = 'rollup-gjs-plugin';
+
+const processor = new Preprocessor();
+// import { parse as pathParse } from 'path';
+
+export default function rollupGjsPlugin(): Plugin {
+  return {
+    name: PLUGIN_NAME,
+    async resolveId(source: string, importer: string | undefined, options) {
+      let resolution = await this.resolve(source, importer, {
+        skipSelf: true,
+        ...options,
+      });
+
+      if (resolution) {
+        return maybeRewriteGJS(resolution);
+      }
+    },
+
+    load(id: string) {
+      const meta = getMeta(this, id);
+      if (!meta) {
+        return;
+      }
+
+      this.addWatchFile(meta.originalId);
+      let input = readFileSync(meta.originalId, 'utf8');
+      let code = processor.process(input);
+      return {
+        code,
+      };
+    },
+  };
+}
+
+type Meta = {
+  originalId: string;
+};
+
+function getMeta(context: PluginContext, id: string): Meta | null {
+  const meta = context.getModuleInfo(id)?.meta?.[PLUGIN_NAME];
+  if (meta) {
+    return meta as Meta;
+  } else {
+    return null;
+  }
+}
+
+const gjsFilter = createFilter('**/*.gjs');
+
+function maybeRewriteGJS(resolution: ResolvedId) {
+  if (!gjsFilter(resolution.id)) {
+    return null;
+  }
+
+  // This creates an `*.gjs.js` that we will populate in `load()` hook.
```

Comment:
This comment appears to be lying, yes? It doesn't turn foo.gjs into foo.gjs.js, it turns it into foo.js. 

I think the code is correct and it's just the comment that is misleading. We don't need to name the files foo.gjs.js. That was more important for foo.hbs.js because that needs to play nice beside foo.js.